firmware/ci.c

```diff
@@ -404,23 +419,161 @@ static void video_mode_list(void)
 
 static void video_mode_set(int mode)
 {
-	char mode_descriptors[PROCESSOR_MODE_COUNT*PROCESSOR_MODE_DESCLEN];
+	char mode_descriptor[PROCESSOR_MODE_DESCLEN];
```

Comment:
When setting a video mode it will print out a message to say what the new video mode is. To get the description, it was generating a description for all video timing modes, but only using the description of the mode that was just set. I changed it to only generate the description for a single mode, rather than doing it for all modes unnecessarily.

If you list the video mode, it will still print out the description for all of the modes as it used to do.